### PR TITLE
Invert reference type null checks

### PR DESCRIPTION
**Pull request by surma**
- [x] I've read the contributing guidelines

(This is my first PR digging into the compiler and I’m not 100% sure I did this right. Please point out anything you’d like changed.)

I was playing with reference types and notices that code like this behaves exactly inverted:

```ts
if(!someExternref) {
  // This is executed when `someExternref` is *not* `null`
}
```

It seems like the logic in `makeIsTrueish` was just using `ref.is_null`, which would explain the inversion. This PR flips thats logic.

However, I couldn’t run the full tests as the tests still failed with `Feature 'gc' is not enabled.` despite setting `ASC_FEATURES="*"`. I guess that is the world of experimental features.

PTAL!

### CODE REVIEW

**Comment by surma** [comment]
Accidentally submitted this in the middle of writing the description. Now ready to review :)

**Comment by SebastianSpeitel** [comment]
> Accidentally submitted this in the middle of writing the description. Now ready to review :)

GitHub has draft pull requests.

File: src/compiler.ts [hunk 1]
```diff
@@ -9941,7 +9941,12 @@ export class Compiler extends DiagnosticEmitter {
       case TypeKind.EQREF:
       case TypeKind.DATAREF:
       case TypeKind.I31REF: {
-        return module.ref_is(RefIsOp.RefIsNull, expr);
+        // Needs to be true when the ref is _not_ null.
+        // Therefore: 1 - <ref.is_null>
+        return type.size == 64
+          ? module.binary(BinaryOp.SubI64, module.i64(1), module.ref_is(RefIsOp.RefIsNull, expr))
+          : module.binary(BinaryOp.SubI32, module.i32(1), module.ref_is(RefIsOp.RefIsNull, expr));
```
Comment (MaxGraey) [inline src/compiler.ts:9948]:
```suggestion
        return type.size == 64
          ? module.unary(BinaryOp.EqzI64, module.ref_is(RefIsOp.RefIsNull, expr))
          : module.unary(BinaryOp.EqzI32, module.ref_is(RefIsOp.RefIsNull, expr));
```

**Comment by surma** [comment]
> GitHub has draft pull requests.

I know, thank you. I just pressed Cmd+Enter by accident.

File: src/compiler.ts [hunk 2]
```diff
@@ -9941,7 +9941,12 @@ export class Compiler extends DiagnosticEmitter {
       case TypeKind.EQREF:
       case TypeKind.DATAREF:
       case TypeKind.I31REF: {
-        return module.ref_is(RefIsOp.RefIsNull, expr);
+        // Needs to be true when the ref is _not_ null.
+        // Therefore: 1 - <ref.is_null>
```
Comment (MaxGraey) [inline src/compiler.ts:9945]:
```suggestion
        // Therefore: 1 - <ref.is_null>
        // Therefore: !(<ref.is_null>)
```

File: src/compiler.ts [hunk 3]
```diff
@@ -9941,7 +9941,13 @@ export class Compiler extends DiagnosticEmitter {
...
+        // Therefore: !(<ref.is_null>)
+        return type.size == 64
+          ? module.unary(UnaryOp.EqzI64, module.ref_is(RefIsOp.RefIsNull, expr))
+          : module.unary(UnaryOp.EqzI32, module.ref_is(RefIsOp.RefIsNull, expr));
```
Comment (dcodeIO) [inline src/compiler.ts:9949]: Afaik `ref.is_null` returns an `i32`.
```suggestion
        return module.unary(UnaryOp.EqzI32, module.ref_is(RefIsOp.RefIsNull, expr));
```

File: src/compiler.ts [hunk 4]
```diff
@@ -9941,7 +9941,13 @@ export class Compiler extends DiagnosticEmitter {
       case TypeKind.EQREF:
       case TypeKind.DATAREF:
       case TypeKind.I31REF: {
-        return module.ref_is(RefIsOp.RefIsNull, expr);
+        // Needs to be true when the ref is _not_ null.
+        // Therefore: 1 - <ref.is_null>
```
Comment (MaxGraey) [inline src/compiler.ts:9945]: Also it seems I accidentally added this comment line which shouldn't exist here

File: src/compiler.ts (hunk 3 above)
Comment (surma) [reply src/compiler.ts:9949]: ooh yeah you are right

**Comment by dcodeIO** [comment]
Thanks again! :)

**Pull request merged**